### DISCUSSION TITLE
GPIO Relay issue

### DISCUSSION THREAD

**Discussion by dohoangminhquan** [theyosh/TerrariumPI#712] [General]
hello everyone,

I'm trying to add a GPIO relay and unable to have it work. I attached a screenshot to this discussion. I did ssh to my Pi and I'm sure that my python code can control the GPIO relay with the same address.

screenshot here https://www.dropbox.com/s/6xgst5sqm9r7r5a/relay.png?dl=0
Anyone please advise me how to resolve the problem ?

thanks,

Quan

**Comment by theyosh** [comment]
Hi,

address 3 is not possible. That is reserved for other stuff (I2C): https://pinout.xyz/pinout/pin3_gpio2#

So, delete this GPIO relay first. Then stop TP4 and power down the Pi. Then reconnect the GPIO relay to another pin: https://pinout.xyz/# use a pin that is green, and does not have a special feature. That is enclosed between '(' and ')'. For example: https://pinout.xyz/pinout/pin11_gpio17# which is address 11.

**Reply by dohoangminhquan** [reply]
Thank you, I made it. However, I turn ON then I got notification "off" and vice versa. Do I need to do anything to correct it? you can see the screenshot here https://www.dropbox.com/s/e8b88kx3tm40uvs/turn%20on%20relay.png?dl=0

thanks

**Reply by dohoangminhquan** [reply]
I restarted my Pi and it works :D

**Reply by theyosh** [reply]
That is strange... no clue why... But lucky it is working now.
